### PR TITLE
[AUTO] Enable AUTO compiledModel::get_property supporting its properties only.

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Enable AUTO compiledModel::get_property supporting its properties only

### Tickets:
 - 96206

### CODE REVIEW

File: src/plugins/auto/auto_executable_network.cpp [hunk 1]
```diff
@@ -175,11 +175,15 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
             }
         }
         return decltype(ov::available_devices)::value_type {exeDevices};
+    } else if (name == ov::model_name) {
+        if (_autoSchedule->_loadContext[ACTUALDEVICE].future.valid()) {
+            _autoSchedule->_loadContext[ACTUALDEVICE].future.wait();
+        }
```
Comment (songbell) [inline src/plugins/auto/auto_executable_network.cpp:181]: I think we don't need to wait for this future

File: src/plugins/auto/auto_executable_network.cpp [hunk 2]
```diff
@@ -177,11 +178,25 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
...
+    } else if (name == METRIC_KEY(SUPPORTED_METRICS)) {
+        IE_SET_METRIC_RETURN(SUPPORTED_METRICS, {
+            METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS),
+            METRIC_KEY(SUPPORTED_METRICS),
+            METRIC_KEY(NETWORK_NAME),
+            METRIC_KEY(SUPPORTED_CONFIG_KEYS)
+        });
+    } else if (name == METRIC_KEY(SUPPORTED_CONFIG_KEYS)) {
+        std::vector<std::string> configKeys = {IE::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES};
+        IE_SET_METRIC_RETURN(SUPPORTED_CONFIG_KEYS, configKeys);
```
Comment (peterchen-intel) [inline src/plugins/auto/auto_executable_network.cpp:198]: Align with the beginning of this function, all RO properties are metrics.

File: src/plugins/auto/auto_executable_network.cpp [hunk 3]
```diff
@@ -26,13 +27,6 @@ void AutoExecutableNetwork::SetConfig(const std::map<std::string, IE::Parameter>
...
 
 IE::Parameter AutoExecutableNetwork::GetConfig(const std::string& name) const {
-    {
-        std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
-        auto it = _autoSContext->_config.find(name);
-        if (it != _autoSContext->_config.end()) {
-            return it->second;
-        }
-    }
     IE_THROW(NotFound) << name << " not found in the ExecutableNetwork config";
```
Comment (peterchen-intel) [inline src/plugins/auto/auto_executable_network.cpp:30]: Is this logic used by MULTI too? @yangwang201911  Will we keep MULTI logic no changed? @wangleis

File: src/plugins/auto/auto_executable_network.cpp [hunk 4]
```diff
@@ -177,11 +171,17 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
...
         }
         return decltype(ov::available_devices)::value_type {exeDevices};
+    } else if (name == ov::model_name) {
+        return _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetMetric(name);
+    } else if (name == METRIC_KEY(SUPPORTED_METRICS)) {
+        IE_SET_METRIC_RETURN(SUPPORTED_METRICS,
+                             {METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS),
+                              METRIC_KEY(SUPPORTED_METRICS),
+                              METRIC_KEY(NETWORK_NAME),
+                              METRIC_KEY(SUPPORTED_CONFIG_KEYS)});
```
Comment (songbell) [inline src/plugins/auto/auto_executable_network.cpp:181]: what is the value of supported_config_keys?

File: src/plugins/auto/auto_executable_network.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/plugins/auto/auto_executable_network.cpp]: Updated.

File: src/plugins/auto/auto_executable_network.cpp (hunk 3 above)
Comment (wangleis) [reply src/plugins/auto/auto_executable_network.cpp:30]: @yangwang201911 @peterchen-intel We will keep the MULTI logic unchanged

File: src/plugins/auto/auto_executable_network.cpp (hunk 4 above)
Comment (yangwang201911) [reply src/plugins/auto/auto_executable_network.cpp:181]: "SUPPORTED_CONFIG_KEYS"
Refer to this line
https://github.com/yangwang201911/openvino/blob/8223969ed00f6df6977b24965a84be8bfaf1cd87/src/inference/include/ie/ie_plugin_config.hpp#L82

File: src/plugins/auto/auto_executable_network.cpp [hunk 5]
```diff
@@ -177,11 +171,17 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
             }
         }
         return decltype(ov::available_devices)::value_type {exeDevices};
+    } else if (name == ov::model_name) {
+        return _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetMetric(name);
+    } else if (name == METRIC_KEY(SUPPORTED_METRICS)) {
+        IE_SET_METRIC_RETURN(SUPPORTED_METRICS,
```
Comment (ilyachur) [inline src/plugins/auto/auto_executable_network.cpp:177]: Do we need to add new or modify existed tests?

Comment (yangwang201911) [reply src/plugins/auto/auto_executable_network.cpp:177]: Updated.

File: src/plugins/auto/auto_executable_network.cpp (hunk 3 above)
Comment (nkogteva) [reply src/plugins/auto/auto_executable_network.cpp:30]: > @yangwang201911 @peterchen-intel We need to keep the MULTI logic unchanged

Why MULTI logic should be unchanged? Don't we need to align behavior across all meta plugins? Or I was missing something

File: src/tests/functional/plugin/shared/include/behavior/executable_network/exec_network_base.hpp [hunk 6]
```diff
@@ -70,6 +70,10 @@ TEST_P(ExecutableNetworkBaseTest, checkGetMetric) {
 
 TEST_P(ExecutableNetworkBaseTest, canLoadCorrectNetworkToGetExecutableAndCheckConfig) {
     auto execNet = ie->LoadNetwork(cnnNet, target_device, configuration);
+    if (target_device == CommonTestUtils::DEVICE_AUTO) {
+        // AUTO executable network didn't support to read any config.
```
Comment (songbell) [inline src/tests/functional/plugin/shared/include/behavior/executable_network/exec_network_base.hpp:74]: probably EXPECT_THROW?

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/include/behavior/executable_network/exec_network_base.hpp:74]: Due to update of supported property list by `AutoExecutableModel `, which means we need to check if the property in the configuration passed from `LoadNetwork()` is supported by `AutoExecutableModel` and then decide to call `EXPECT_THROW `or `ASSERT_NO_THROW `to complete this test case. Actual the test case of `AutoExecutableModel` supported properties checking has been implemented in `src/tests/functional/plugin/shared/src/behavior/ov_executable_network/get_metric.cpp`

File: src/plugins/auto/auto_executable_network.cpp (hunk 3 above)
Comment (riverlijunjie) [reply src/plugins/auto/auto_executable_network.cpp:30]: Need we align the same behavior between SetConfig and GetConfig?

**Review by wangleis — APPROVED** [review]

**Pull request merged**